tools/backlog-groomer/tests/test_report.py

```
@@ -0,0 +1,94 @@
+    assert "| armed-stuck | not evaluated |" in report
+
+
+def test_titles_are_defused_and_capped():
+    nasty = "A `weird` [title] | with *markdown*\nand a newline " + "x" * 100
+    snapshot = _mini_snapshot([
+        {"number": 10, "title": nasty, "labels": [],
+         "createdAt": "2026-07-01T06:00:00+00:00", "updatedAt": "2026-07-01T06:00:00+00:00"},
+    ])
+    cfg = Config()
+    report = render(evaluate(snapshot, cfg), snapshot, cfg)
+    line = next(l for l in report.splitlines() if l.startswith("- #10"))
```

Ruff (0.16.1)

[error] Ambiguous variable name: `l`

(E741)